### PR TITLE
Enforce script naming conventions on new levelbuilder scripts

### PR DESCRIPTION
**Pull request by joshlory**
New scripts created on levelbuilder can only use characters `[a-z0-9\-]` in the name. Existing scripts with "invalid" names can still be seeded and edited on levelbuilder.

### CODE REVIEW

**Comment by joshlory** [comment]
FYI @mehalshah.

**Comment by joshlory** [comment]
Sample error message:

script-name-format-validation-failed

**Comment by wjordan** [comment]
LGTM! good idea, this will help prevent a lot of annoying edge-cases in weird script names.

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -35,7 +35,17 @@ class Script < ActiveRecord::Base
...
-  validates :name, presence: true, uniqueness: { case_sensitive: false}
+
+  attr_accessor :skip_name_format_validation
+
+  validates :name,
+    presence: true,
+    uniqueness: {case_sensitive: false},
+    format: {
+      unless: :skip_name_format_validation,
+      with: /\A[a-z0-9\-]+\z/,
```
Comment (mehalshah) [inline dashboard/app/models/script.rb:46]: Does this get us into trouble with non ASCII characters or characters with accents? Is é valid or invalid in a script name?

**Comment by mehalshah** [comment]
lgtm

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (joshlory) [inline dashboard/app/models/script.rb:46]: Hmm, I think we only want to allow `a-z` letters.  Users may have trouble typing "é" in the address bar.

Comment (joshlory) [inline dashboard/app/models/script.rb:46]: There's also a localized title for each script in `scripts.[locale].yml`.  The `:name` field only corresponds to the URL and filename.

**Pull request merged**